packages/report-server/src/reportBuilder/functions/basic.ts

```diff
@@ -1,11 +1,19 @@
-import { FieldValue } from '../reportBuilder';
+import { FieldValue } from '../types';
 
 export const value = (valueGiven: FieldValue): FieldValue => {
   return valueGiven;
 };
 
 export const last = (values: FieldValue[]): FieldValue => {
-  return values.reverse()[0];
+  if (!Array.isArray(values)) {
+    throw new Error(`Function 'last' expected an array, but got: ${values}`);
+  }
+
+  if (values.length < 1) {
+    return undefined;
+  }
+
+  return values[values.length - 1];
```

Comment:
Pretty funny bug here, we were previously reversing the same array each time this was called, which lead to it returning last, then first, then last, etc... 😝